From 61da06a2939d479a42c7f48ed5abb642124ada1a Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Tue, 22 Feb 2022 10:38:13 +0000
Subject: [PATCH] Latest clang is causing build failures because -Werror is
 used:

  edk2/MdeModulePkg/Include/Guid/ExtendedFirmwarePerformance.h:78:47:
  error: field Guid within 'FPDT_GUID_EVENT_RECORD' is less aligned than 'EFI_GUID'
  and is usually due to 'FPDT_GUID_EVENT_RECORD' being packed, which can lead to
  unaligned accesses [-Werror,-Wunaligned-access]

This has been reported upstream[1] so until this is resolved, ignore the warnings.

[1] https://edk2.groups.io/g/devel/message/86838

Upstream-Status: Pending
Signed-off-by: Ross Burton <ross.burton@arm.com>

---
 BaseTools/Conf/tools_def.template | 2 +-
 ShellPkg/Application/sbsa-acs     | 1 +
 edk2-libc                         | 1 +
 edk2-platforms                    | 1 +
 4 files changed, 4 insertions(+), 1 deletion(-)
 create mode 160000 ShellPkg/Application/sbsa-acs
 create mode 160000 edk2-libc
 create mode 160000 edk2-platforms

diff --git a/BaseTools/Conf/tools_def.template b/BaseTools/Conf/tools_def.template
index f2bb6247e8..ca2b449f0a 100755
--- a/BaseTools/Conf/tools_def.template
+++ b/BaseTools/Conf/tools_def.template
@@ -2567,7 +2567,7 @@ DEFINE CLANG38_X64_PREFIX           = ENV(CLANG38_BIN)
 DEFINE CLANG38_IA32_TARGET          = -target i686-pc-linux-gnu
 DEFINE CLANG38_X64_TARGET           = -target x86_64-pc-linux-gnu
 
-DEFINE CLANG38_WARNING_OVERRIDES    = -Wno-parentheses-equality -Wno-tautological-compare -Wno-tautological-constant-out-of-range-compare -Wno-empty-body -Wno-unused-const-variable -Wno-varargs -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-const-variable
+DEFINE CLANG38_WARNING_OVERRIDES    = -Wno-parentheses-equality -Wno-tautological-compare -Wno-tautological-constant-out-of-range-compare -Wno-empty-body -Wno-unused-const-variable -Wno-varargs -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-const-variable -Wno-error=unaligned-access
 DEFINE CLANG38_ALL_CC_FLAGS         = DEF(GCC48_ALL_CC_FLAGS) DEF(CLANG38_WARNING_OVERRIDES) -fno-stack-protector -mms-bitfields -Wno-address -Wno-shift-negative-value -Wno-unknown-pragmas -Wno-incompatible-library-redeclaration -fno-asynchronous-unwind-tables -mno-sse -mno-mmx -msoft-float -mno-implicit-float  -ftrap-function=undefined_behavior_has_been_optimized_away_by_clang -funsigned-char -fno-ms-extensions -Wno-null-dereference
 
 ###########################
diff --git a/ShellPkg/Application/sbsa-acs b/ShellPkg/Application/sbsa-acs
new file mode 160000
index 0000000000..ec02a7736a
--- /dev/null
+++ b/ShellPkg/Application/sbsa-acs
@@ -0,0 +1 @@
+Subproject commit ec02a7736ae5714326507c60595f4d5299e3dec8
diff --git a/edk2-libc b/edk2-libc
new file mode 160000
index 0000000000..61687168fe
--- /dev/null
+++ b/edk2-libc
@@ -0,0 +1 @@
+Subproject commit 61687168fe02ac4d933a36c9145fdd242ac424d1
diff --git a/edk2-platforms b/edk2-platforms
new file mode 160000
index 0000000000..c9e377b00f
--- /dev/null
+++ b/edk2-platforms
@@ -0,0 +1 @@
+Subproject commit c9e377b00fc086fcb5a5b41663a0149bde9bcc2e
